src/screen/CategoryPage.tsx

```diff
@@ -37,28 +37,39 @@ const defaultInputCategory: CategoryForm = {
 };
 
 export function CategoryPage(_props: CategoryPageProps) {
-  const [inputCategory, setInputCategory] =
-    useState<CategoryForm>(defaultInputCategory);
-  const fetchCategories = useFetchCategories();
-  const createCategory = useCreateCategory();
   const deleteCategory = useDeleteCategory();
   const editCategory = useEditCategory();
+  const mengapa = categoryhooks();
 
-  function onInputCategory(props: CategoryForm) {
+  async function onInputCategory(props: CategoryForm) {
     if (props.type === "add") {
-      createCategory.submit({ title: props.title }).then(() => {
-        setInputCategory(defaultInputCategory);
-        fetchCategories.reFetch();
-      });
+      await mengapa.useCreateCategory.mutate({ title: props.title });
+      await inputCategory.setValues(defaultInputCategory);
+      inputCategory.setErrors({});
+      mengapa.useFetchCategory.refetch();
```

Comment:
Kalau gak mau pake "async" bisa kayak gini

```suggestion
      mengapa.useCreateCategory.mutate({ title: props.title }).then(() => {
        
        // formik perlu dibuat "untouched" dulu, sesuaikan dengan API nya apa :
        inputCategory.setTouched({...})
        
        inputCategory.setValues(defaultInputCategory);
        inputCategory.setErrors({});
        mengapa.useFetchCategory.refetch();
      });
```

Reply:
done update